src/bson.c

```diff
@@ -1401,8 +1406,12 @@ PHONGO_API int phongo_bson_to_zval_ex(const unsigned char *data, int data_len, p
 	if (bson_iter_visit_all(&iter, &php_bson_visitors, state) || iter.err_off) {
 		/* Iteration stopped prematurely due to corruption or a failed visitor.
 		 * While we free the reader, state->zchild should be left as-is, since
-		 * the calling code may want to zval_ptr_dtor() it. */
-		phongo_throw_exception(PHONGO_ERROR_UNEXPECTED_VALUE TSRMLS_CC, "Could not convert BSON document to a PHP variable");
+		 * the calling code may want to zval_ptr_dtor() it. If an exception has
+		 * been thrown already (due to an unsupported BSON type for example,
+		 * don't overwrite with a generic exception message. */
+		if (!EG(exception)) {
+			phongo_throw_exception(PHONGO_ERROR_UNEXPECTED_VALUE TSRMLS_CC, "Detected corrupt BSON data.");
```

Comment:
I realize you used the same message in https://github.com/mongodb/mongo-hhvm-driver/commit/1d51fe96e2fdc24596f4a47affbfbbfb1f34c239, but should there really be a period after this message? We typically don't use that, and only did so in the corruption message since it consists of two sentences.